hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/webapp/TestRMWebServicesCapacitySchedDynamicConfig.java

```diff
@@ -18,311 +18,161 @@
+    conf.put("yarn.scheduler.capacity.root.default.capacity", "4w");
+    conf.put("yarn.scheduler.capacity.root.test1.capacity", "16w");
+    conf.put("yarn.scheduler.capacity.root.test2.capacity", "12w");
+    conf.put("yarn.scheduler.capacity.root.test1.test1_1.capacity", "2w");
+    conf.put("yarn.scheduler.capacity.root.test1.test1_2.capacity", "2w");
+    conf.put("yarn.scheduler.capacity.root.test1.test1_3.capacity", "12w");
+    MockRM rm = createMutableRM(createConfiguration(conf));
+    runTest(EXPECTED_FILE_TMPL, "testWeightMode", rm, resource());
   }
 
   @Test
-  public void testSchedulerResponseWeightModeWithAutoCreatedQueues()
-      throws Exception {
-    Configuration config = CSConfigGenerator
-        .createWeightConfigWithAutoQueueCreationEnabled();
-    config.setInt(CapacitySchedulerConfiguration
-        .getQueuePrefix("root.autoParent1") +
-        AutoCreatedQueueTemplate.AUTO_QUEUE_TEMPLATE_PREFIX +
-        "maximum-applications", 300);
-
-    initResourceManager(config);
-    initAutoQueueHandler(1200 * GB);
-
-    // same as webapp/scheduler-response-WeightMode.json, but with effective resources filled in
-    assertJsonResponse(sendRequest(),
-        "webapp/scheduler-response-WeightModeWithAutoCreatedQueues-Before.json");
-
-    //Now create some auto created queues
-    createQueue("root.auto1");
-    createQueue("root.auto2");
-    createQueue("root.auto3");
-    createQueue("root.autoParent1.auto4");
-    createQueue("root.autoParent2.auto5");
-    createQueue("root.parent.autoParent3.auto6");
-
-    assertJsonResponse(sendRequest(),
-        "webapp/scheduler-response-WeightModeWithAutoCreatedQueues-After.json");
-  }
-
-  private void initAutoQueueHandler(int nodeMemory) throws Exception {
-    CapacityScheduler cs = (CapacityScheduler) rm.getResourceScheduler();
-    autoQueueHandler = cs.getCapacitySchedulerQueueManager();
-    rm.registerNode("h1:1234", nodeMemory); // label = x
+  public void testWeightModeFlexibleAQC() throws Exception {
+    Map<String, String> conf = new HashMap<>();
+    conf.put("yarn.scheduler.capacity.root.queues", "default, test1, test2");
+    conf.put("yarn.scheduler.capacity.root.test1.queues", "test1_1, test1_2, test1_3");
+    conf.put("yarn.scheduler.capacity.root.default.capacity", "4w");
+    conf.put("yarn.scheduler.capacity.root.test1.capacity", "16w");
+    conf.put("yarn.scheduler.capacity.root.test2.capacity", "12w");
+    conf.put("yarn.scheduler.capacity.root.test1.test1_1.capacity", "2w");
+    conf.put("yarn.scheduler.capacity.root.test1.test1_2.capacity", "2w");
+    conf.put("yarn.scheduler.capacity.root.test1.test1_3.capacity", "12w");
+
+    Configuration config = createConfiguration(conf);
+    setupAQC(config, "yarn.scheduler.capacity.root.test1.");
+    setupAQC(config, "yarn.scheduler.capacity.root.test1.test1_1.");
+    try (MockRM rm = createMutableRM(config)) {
+      createAQC(rm, "test1");
```

Comment:
first register the node

then assertjson with a before-aqc suffix

then create the queues

then assertjson with an after-aqc suffix

So we can compare the two jsons.